client/controllers/graph.controller.js

```diff
@@ -1,16 +1,85 @@
 (function() {
-    
     "use strict";
     
     angular
-        .module("graph", [])
+        .module("smitty.graph", [])
         .controller("GraphController", GraphController);
     
-    function GraphController($scope, $http){ 
+    GraphController.$inject = ["canvasService"];
+    
+    function GraphController(canvasService) {
+        var vm = this; // View Model
+        
+        var nextVertexId = 0;
+        var vertexColor = "#CC8193";
+        
+        // Set of vertices in graph - V(G)
+        vm.vertices = [
+            {
+                id: 0, // id of vertex
+                xPos: 0, // x-position
+                yPos: 0, // y-position
+                weight: 0, // vertex weight (if applicable)
+                size: 20 // size of vertex (radius)
+            }
+        ];
         
+        // Set of edges in graph - E(G)
+        vm.edges = [
+            {
+                // TODO
+            }
+        ];
         
+        // Function: addVertex()
+        // adds vertex to V(G)
+        vm.addVertex = function(event) {
+            vm.offsetX = event.offsetX;
+            vm.offsetY = event.offsetY;
+            vm.id = nextVertexId;
+            
+            vm.vertices.push({
+                id: vm.id,
+                xPos: vm.offsetX,
+                yPos: vm.offsetY,
+                weight: 0,
+                size: 10
+            });
+            nextVertexId++;
```

Comment:
This and line 39 are new and relate to my comment about IDs above. When you start pulling this from a db, line 48 will become obsolete.